src/WebJobs.Script/Workers/Rpc/Configuration/ProfileCondition.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc
+{
+    public enum ConditionType
+    {
+        [EnumMember(Value = "environment")]
+        Environment,
+        [EnumMember(Value = "hostProperty")]
+        HostProperty
+    }
+
+    public enum ConditionHostPropertyName
+    {
+        [EnumMember(Value = "sku")]
```

Comment:
These would ideally all be encapsulated by the "HostEnvironmentCondition" implementation.